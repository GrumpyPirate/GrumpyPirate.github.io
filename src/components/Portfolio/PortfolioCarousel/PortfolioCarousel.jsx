import React from 'react'
import PropTypes from 'prop-types'

// Components
import Carousel from 'components/Carousel/Carousel.jsx'

// Style
import './PortfolioCarousel.scss'

// images
import frameMobile from 'images/device-frames/frame-mobile.svg'
import frameTablet from 'images/device-frames/frame-tablet.svg'
import frameLaptop from 'images/device-frames/frame-laptop.svg'

// Settings
import { animation } from 'config/animation'

const PortfolioCarousel = ({ desktop, tablet, mobile }) => {
  return (
    <div className="pf-carousel">
      <Carousel showStatus={false} showThumbs={false} infiniteLoop={true} useKeyboardArrows={true} transitionTime={animation.duration * 2} dynamicHeight showArrows={false} emulateTouch swipeScrollTolerance={75}>
          {mobile &&
            <div className="pf-carousel__slide pf-carousel__slide--mobile">
              <figure className="pf-carousel__figure pf-carousel__figure--mobile">
                <img src={frameMobile} className="pf-carousel__device-frame"/>
                <div className="pf-carousel__device-img pf-carousel__device-img--mobile">
                  <img src={mobile} alt="" />
                </div>
              </figure>
            </div>
          }
          {tablet &&
            <div className="pf-carousel__slide pf-carousel__slide--tablet">
              <figure className="pf-carousel__figure pf-carousel__figure--tablet">
                <img src={frameTablet} className="pf-carousel__device-frame"/>
                <div className="pf-carousel__device-img pf-carousel__device-img--tablet">
                  <img src={tablet} alt="" />
                </div>
              </figure>
            </div>
          }
          {desktop &&
            <div className="pf-carousel__slide pf-carousel__slide--desktop">
              <figure className="pf-carousel__figure pf-carousel__figure--desktop">
                <img src={frameLaptop} className="pf-carousel__device-frame"/>
                <div className="pf-carousel__device-img pf-carousel__device-img--desktop">
                  <img src={desktop} alt="" />
                </div>
              </figure>
            </div>
          }
      </Carousel>
    </div>
  )
} // /const PortfolioCarousel

PortfolioCarousel.propTypes = {
  desktop: PropTypes.node,
  tablet: PropTypes.node,
  mobile: PropTypes.node
} // /PortfolioCarousel.propTypes

export default PortfolioCarousel
