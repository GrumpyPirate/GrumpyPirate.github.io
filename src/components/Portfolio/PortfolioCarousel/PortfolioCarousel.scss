@import 'core';

.pf-carousel {
  // Slides
  // &__slide {
  // } // /&__slide

  &__figure {
    position: relative;

    // Figure variants
    &--desktop {
      $aspect: (291 / 506);
      $device-img-width-diff: .75;
      $device-img-height-diff: .8125;
      $device-img-offset-top: .062;
      $device-img-offset-left: .125;
      $device-frame-width-xs: 290;
      $device-frame-width-md: 480;
      $device-frame-width-lg: 640;

      .pf-carousel__device-frame {
        width: rem($device-frame-width-xs * 1px);
        height: rem($device-frame-width-xs * $aspect * 1px);
      } // /.pf-carousel__device-frame

      .pf-carousel__device-img {
        width: rem($device-frame-width-xs * $device-img-width-diff * 1px);
        height: rem($device-frame-width-xs * $aspect * $device-img-height-diff * 1px);
        top: rem($device-frame-width-xs * $device-img-offset-top * 1px);
        left: rem($device-frame-width-xs * $device-img-offset-left * 1px);
      } // /.pf-carousel__device-img

      // md up
      @include media-breakpoint-up(md) {
        .pf-carousel__device-frame {
          width: rem($device-frame-width-md * 1px);
          height: rem($device-frame-width-md * $aspect * 1px);
        } // /.pf-carousel__device-frame

        .pf-carousel__device-img {
          width: rem($device-frame-width-md * $device-img-width-diff * 1px);
          height: rem($device-frame-width-md * $aspect * $device-img-height-diff * 1px);
        } // /.pf-carousel__device-frame
      } // /md up
    } // /&--desktop

    &--tablet {

    } // /&--tablet

    &--mobile {

    } // /&--mobile
  } // /&__figure

  // Device frames (laptop, tablet, mobile)
  &__device-frame {
    position: relative;
    display: block;
    max-width: 100%;
    margin: auto;
    z-index: 1;
  } // /&__device-frame

  // Device images
  &__device-img {
    position: absolute;
    overflow-y: auto;
    z-index: 2;

    img {
      display: block;
      width: 100%;
      height: auto;
    } // /img
  } // /&__device-frame
} // /.pf-carousel
