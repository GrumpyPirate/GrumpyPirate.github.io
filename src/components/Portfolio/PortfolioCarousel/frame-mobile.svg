<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 222 439"><defs><style>.a{fill:none;stroke:#231f20;stroke-width:2px;}</style></defs><title>mobile-frame</title><rect class="a" x="1" y="1" width="220" height="437" rx="34" ry="34"/><rect class="a" x="14" y="56" width="194" height="330"/><circle class="a" cx="111" cy="411" r="14"/></svg>