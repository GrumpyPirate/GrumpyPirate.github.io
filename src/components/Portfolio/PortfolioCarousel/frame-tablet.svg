<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 322 477"><defs><style>.a{fill:none;stroke:#231f20;stroke-width:2px;}</style></defs><title>tablet-frame_1</title><rect class="a" x="1" y="1" width="320" height="475" rx="25" ry="25"/><rect class="a" x="17.5" y="44" width="288" height="382.5"/><circle class="a" cx="161.5" cy="451.5" r="14"/></svg>