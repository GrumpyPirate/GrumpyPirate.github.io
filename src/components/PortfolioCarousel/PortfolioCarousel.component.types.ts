export interface PortfolioCarouselProps {
  desktopImage: string;
  tabletImage: string;
  mobileImage: string;
}
