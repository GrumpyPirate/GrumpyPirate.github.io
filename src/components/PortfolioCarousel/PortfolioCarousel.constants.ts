import styled, { css, SimpleInterpolation } from 'styled-components';

import { animDur, mediaQueries, overflowYScroll, palette, rem } from 'styles';

type SlideVariant = 'mobile' | 'tablet' | 'desktop';

const mobileAspectRatio = 439 / 222;
const mobileDeviceImgWidthDiff = 0.8675;
const mobileDeviceImgHeightDiff = 0.74725;
const mobileDeviceImgOffsetTop = 0.2575;
const mobileDeviceFrameWidthXs = 192;
const mobileDeviceFrameWidthSm = 192;

const tabletAspectRatio = 477 / 322;
const tabletDeviceImgWidthDiff = 0.8875;
const tabletDeviceImgHeightDiff = 0.80125;
const tabletDeviceImgOffsetTop = 0.1375;
const tabletDeviceFrameWidthXs = 224;
const tabletDeviceFrameWidthSm = 320;

const desktopAspectRatio = 291 / 506;
const desktopDeviceImgWidthDiff = 0.75;
const desktopDeviceImgHeightDiff = 0.81375;
const desktopDeviceImgOffsetTop = 0.036225;
const desktopDeviceFrameWidthXs = 290;
const desktopDeviceFrameWidthSm = 420;

export const carouselStyles = css`
  .carousel {
    position: relative;
    width: 100%;

    * {
      box-sizing: border-box;
      padding: 0;
    }

    button {
      background: none;
      border: 0;
      outline: 0;
    }

    img {
      pointer-events: none;
    }

    .carousel {
      position: relative;
    }

    .control-arrow {
      font-size: 18px;
      margin-top: -13px;
      top: 50%;
    }

    .thumbs-wrapper {
      margin: 20px;
      overflow: hidden;
    }

    .thumbs {
      list-style: none;
      position: relative;
      transform: translate3d(0, 0, 0);
      transition: all 0.15s ease-in;
      white-space: nowrap;
    }

    .thumb {
      border: 3px solid #fff;
      display: inline-block;
      margin-right: 6px;
      overflow: hidden;
      padding: 2px;
      transition: border 0.15s ease-in;
      white-space: nowrap;
      width: 80px;

      &.selected,
      &:hover {
        border: 3px solid #333;
        padding: 2px;
      }

      img {
        vertical-align: top;
      }
    }

    .control-dots {
      margin: 0.5rem 0;
      text-align: center;

      .dot {
        background: ${palette.themeAccentLight};
        border-radius: 50%;
        cursor: pointer;
        display: inline-block;
        height: 8px;
        margin: 0 8px;
        opacity: 0.5;
        transition-duration: ${animDur}s;
        transition-property: border, opacity, transform, background-color;
        width: 8px;

        &:hover,
        &.selected {
          background-color: transparent;
          border: solid 1px ${palette.themeAccentLight};
          transform: scale(2);
        }

        &.selected {
          opacity: 1;
        }
      }
    }
  }

  .carousel .control-arrow,
  .carousel.carousel-slider .control-arrow {
    background: none;
    border: 0;
    cursor: pointer;
    font-size: 32px;
    opacity: 0.4;
    position: absolute;
    top: 20px;
    transition: all 0.25s ease-in;
    z-index: 2;

    .control-arrow {
      &:hover {
        opacity: 1;
      }

      &::before {
        border-bottom: 8px solid transparent;
        border-top: 8px solid transparent;
        content: '';
        display: inline-block;
        margin: 0 5px;
      }
    }
  }

  .carousel .control-disabled.control-arrow {
    cursor: inherit;
    display: none;
    opacity: 0;
  }

  .carousel .control-prev.control-arrow {
    left: 0;
  }

  .carousel .control-prev.control-arrow::before {
    border-right: 8px solid #fff;
  }

  .carousel .control-next.control-arrow {
    right: 0;
  }

  .carousel .control-next.control-arrow::before {
    border-left: 8px solid #fff;
  }

  .carousel.carousel-slider {
    margin: 0;
    overflow: hidden;
    position: relative;
  }

  .carousel.carousel-slider .control-arrow {
    bottom: 0;
    color: #fff;
    font-size: 26px;
    margin-top: 0;
    padding: 5px;
    top: 0;
  }

  .carousel.carousel-slider .control-arrow:hover {
    background: rgba(0, 0, 0, 0.2);
  }

  .carousel .slider-wrapper {
    margin: auto;
    overflow: hidden;
    transition: height 0.15s ease-in;
    width: 100%;
  }

  .carousel .slider-wrapper.axis-horizontal .slider {
    display: flex;
  }

  .carousel .slider-wrapper.axis-horizontal .slider .slide {
    flex-direction: column;
    flex-flow: column;
  }

  .carousel .slider-wrapper.axis-vertical {
    display: flex;
  }

  .carousel .slider-wrapper.axis-vertical .slider {
    flex-direction: column;
  }

  .carousel .slider {
    list-style: none;
    position: relative;
    width: 100%;
  }

  .carousel .slider.animated {
    transition: all 0.35s ease-in-out;
  }

  .carousel .slide {
    margin: 0;
    min-width: 100%;
    position: relative;
    text-align: center;
  }

  .carousel .slide img {
    border: 0;
    vertical-align: top;
  }

  .carousel .slide iframe {
    border: 0;
    display: inline-block;
    margin: 0 40px 40px;
    width: calc(100% - 80px);
  }

  .carousel .slide .legend {
    background: #000;
    border-radius: 10px;
    bottom: 40px;
    color: #fff;
    font-size: 12px;
    left: 50%;
    margin-left: -45%;
    opacity: 0.25;
    padding: 10px;
    position: absolute;
    text-align: center;
    transition: all 0.5s ease-in-out;
    transition: opacity 0.35s ease-in-out;
    width: 90%;
  }

  .carousel .carousel-status {
    color: #fff;
    font-size: 10px;
    padding: 5px;
    position: absolute;
    right: 0;
    text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.9);
    top: 0;
  }

  .carousel:hover .slide .legend {
    opacity: 1;
  }
`;

export const DeviceFrame = styled.img`
  display: block;
  max-width: 100%;
  position: relative;
  user-select: none;
  z-index: 2;
  margin: 0 auto;
`;

export const DevicePreview = styled.figure`
  ${overflowYScroll}
  touch-action: pan-y;
  position: absolute;
  background-color: ${palette.greyDark};
  z-index: 1;

  img {
    display: block;
    height: auto;
    width: 100%;
    margin: 0 auto;
  }
`;

export const SlideImage = styled.figure`
  position: relative;
  margin: 0;
  user-select: none;
`;

const createSlideStyles = ({
  imageAspectRatio,
  deviceImgWidthDiff,
  deviceImgHeightDiff,
  deviceImgOffsetTop,
  deviceFrameWidthXs,
  deviceFrameWidthSm,
}: {
  imageAspectRatio: number;
  deviceImgWidthDiff: number;
  deviceImgHeightDiff: number;
  deviceImgOffsetTop: number;
  deviceFrameWidthXs: number;
  deviceFrameWidthSm: number;
}): SimpleInterpolation => css`
  ${DeviceFrame} {
    height: ${rem(deviceFrameWidthXs * imageAspectRatio)};
    width: ${rem(deviceFrameWidthXs)};
  }

  ${DevicePreview} {
    height: ${rem(deviceFrameWidthXs * imageAspectRatio * deviceImgHeightDiff)};
    left: 0;
    margin: 0 auto;
    right: 0;
    top: ${rem(deviceFrameWidthXs * deviceImgOffsetTop)};
    width: ${rem(deviceFrameWidthXs * deviceImgWidthDiff)};
  }

  @media ${mediaQueries.sm} {
    ${DeviceFrame} {
      height: ${rem(deviceFrameWidthSm * imageAspectRatio)};
      width: ${rem(deviceFrameWidthSm)};
    }

    ${DevicePreview} {
      height: ${rem(deviceFrameWidthSm * imageAspectRatio * deviceImgHeightDiff)};
      top: ${rem(deviceFrameWidthSm * deviceImgOffsetTop)};
      width: ${rem(deviceFrameWidthSm * deviceImgWidthDiff)};
    }
  }
`;

export const Slide = styled.div<{ variant: SlideVariant }>`
  ${({ variant }) => {
    switch (variant) {
      case 'desktop':
        return createSlideStyles({
          imageAspectRatio: desktopAspectRatio,
          deviceFrameWidthXs: desktopDeviceFrameWidthXs,
          deviceFrameWidthSm: desktopDeviceFrameWidthSm,
          deviceImgHeightDiff: desktopDeviceImgHeightDiff,
          deviceImgOffsetTop: desktopDeviceImgOffsetTop,
          deviceImgWidthDiff: desktopDeviceImgWidthDiff,
        });
      case 'tablet':
        return createSlideStyles({
          imageAspectRatio: tabletAspectRatio,
          deviceFrameWidthXs: tabletDeviceFrameWidthXs,
          deviceFrameWidthSm: tabletDeviceFrameWidthSm,
          deviceImgHeightDiff: tabletDeviceImgHeightDiff,
          deviceImgOffsetTop: tabletDeviceImgOffsetTop,
          deviceImgWidthDiff: tabletDeviceImgWidthDiff,
        });
      case 'mobile':
      default:
        return createSlideStyles({
          imageAspectRatio: mobileAspectRatio,
          deviceFrameWidthXs: mobileDeviceFrameWidthXs,
          deviceFrameWidthSm: mobileDeviceFrameWidthSm,
          deviceImgHeightDiff: mobileDeviceImgHeightDiff,
          deviceImgOffsetTop: mobileDeviceImgOffsetTop,
          deviceImgWidthDiff: mobileDeviceImgWidthDiff,
        });
    }
  }}
`;
