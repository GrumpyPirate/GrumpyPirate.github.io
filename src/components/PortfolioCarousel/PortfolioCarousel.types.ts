import { PropsWithClassName } from 'types/common';

export type PortfolioCarouselProps = PropsWithClassName<{
  desktopImage: string;
  tabletImage: string;
  mobileImage: string;
}>;
