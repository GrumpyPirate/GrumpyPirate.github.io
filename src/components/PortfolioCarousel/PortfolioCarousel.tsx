import React, { FunctionComponent } from 'react';
import { Carousel } from 'react-responsive-carousel';
import styled from 'styled-components';

import LoadableImage from 'components/LoadableImage/LoadableImage';
import frameLaptop from 'images/device-frames/frame-laptop.svg';
import frameMobile from 'images/device-frames/frame-mobile.svg';
import frameTablet from 'images/device-frames/frame-tablet.svg';
import { ContentService } from 'services/ContentService';
import { animDur, mediaQueries } from 'styles';

import {
  carouselStyles,
  DeviceFrame,
  DevicePreview,
  Slide,
  SlideImage,
} from './PortfolioCarousel.constants';
import { PortfolioCarouselProps } from './PortfolioCarousel.types';

const PortfolioCarousel: FunctionComponent<PortfolioCarouselProps> = function ({
  className,
  desktopImage,
  tabletImage,
  mobileImage,
}) {
  return (
    <div className={className}>
      <Carousel
        showStatus={false}
        showThumbs={false}
        infiniteLoop
        useKeyboardArrows
        transitionTime={animDur * 2}
        dynamicHeight
        showArrows={false}
        emulateTouch
        swipeScrollTolerance={25}
      >
        <Slide variant="mobile">
          <SlideImage>
            <DeviceFrame src={frameMobile} alt="" />
            <DevicePreview>
              <picture>
                <source
                  type="image/webp"
                  srcSet={`${ContentService.getResizedImage(mobileImage, {
                    format: 'webp',
                    width: 166,
                  })} 166w, ${ContentService.getResizedImage(mobileImage, {
                    format: 'webp',
                    width: 332,
                  })} 332w`}
                  sizes="166px"
                />
                <source
                  type="image/jpeg"
                  srcSet={`${ContentService.getResizedImage(mobileImage, {
                    format: 'jpg',
                    width: 166,
                  })} 166w, ${ContentService.getResizedImage(mobileImage, {
                    format: 'jpg',
                    width: 332,
                  })} 332w`}
                  sizes="166px"
                />

                <LoadableImage loading="lazy" src={mobileImage} alt="" />
              </picture>
            </DevicePreview>
          </SlideImage>
        </Slide>

        <Slide variant="tablet">
          <DeviceFrame src={frameTablet} alt="" />
          <DevicePreview>
            <picture>
              <source
                type="image/webp"
                srcSet={`${ContentService.getResizedImage(tabletImage, {
                  format: 'webp',
                  width: 200,
                })} 200w, ${ContentService.getResizedImage(tabletImage, {
                  format: 'webp',
                  width: 400,
                })} 400w`}
                sizes="200px"
              />
              <source
                type="image/jpeg"
                srcSet={`${ContentService.getResizedImage(tabletImage, {
                  format: 'jpg',
                  width: 200,
                })} 200w, ${ContentService.getResizedImage(tabletImage, {
                  format: 'jpg',
                  width: 400,
                })} 400w`}
                sizes="200px"
              />
              <LoadableImage loading="lazy" src={tabletImage} alt="" />
            </picture>
          </DevicePreview>
        </Slide>

        <Slide variant="desktop">
          <DeviceFrame src={frameLaptop} alt="" />
          <DevicePreview>
            <picture>
              <source
                type="image/webp"
                srcSet={`${ContentService.getResizedImage(desktopImage, {
                  format: 'webp',
                  width: 218,
                })} 218w, ${ContentService.getResizedImage(desktopImage, {
                  format: 'webp',
                  width: 436,
                })} 436w`}
                sizes="218px"
              />
              <source
                type="image/jpeg"
                srcSet={`${ContentService.getResizedImage(desktopImage, {
                  format: 'jpg',
                  width: 218,
                })} 218w, ${ContentService.getResizedImage(desktopImage, {
                  format: 'jpg',
                  width: 436,
                })} 436w`}
                sizes="218px"
              />
              <LoadableImage loading="lazy" src={desktopImage} alt="" />
            </picture>
          </DevicePreview>
        </Slide>
      </Carousel>
    </div>
  );
};

export default styled(PortfolioCarousel)`
  ${carouselStyles}

  @media ${mediaQueries.md} {
    display: none;
  }
`;
