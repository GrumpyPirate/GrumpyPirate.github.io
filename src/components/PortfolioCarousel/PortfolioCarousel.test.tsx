import { render } from '@testing-library/react';
import React from 'react';

import PortfolioCarousel from './PortfolioCarousel';
import fixture from './PortfolioCarousel.fixture';

describe('Components', () => {
  describe('PortfolioCarousel', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const { container } = render(<PortfolioCarousel {...props} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
