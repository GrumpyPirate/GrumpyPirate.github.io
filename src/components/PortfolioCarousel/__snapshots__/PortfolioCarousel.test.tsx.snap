// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PortfolioCarousel should render as expected, without crashing 1`] = `
<div
  className="portfolio-carousel"
>
  <Carousel
    dynamicHeight={true}
    emulateTouch={true}
    infiniteLoop={true}
    showArrows={false}
    showStatus={false}
    showThumbs={false}
    swipeScrollTolerance={75}
    transitionTime={300}
    useKeyboardArrows={true}
  >
    <div
      className="portfolio-carousel__slide portfolio-carousel__slide--mobile"
    >
      <figure
        className="portfolio-carousel__figure portfolio-carousel__figure--mobile"
      >
        <img
          alt=""
          className="portfolio-carousel__device-frame"
          src="frame-mobile.svg"
        />
        <div
          className="portfolio-carousel__device-img portfolio-carousel__device-img--mobile"
        >
          <img
            alt=""
            src="https://maremmatic.com/versation/frigorimeter?a=septocosta&b=blastie#pleasurehood"
          />
        </div>
      </figure>
    </div>
    <div
      className="portfolio-carousel__slide portfolio-carousel__slide--tablet"
    >
      <figure
        className="portfolio-carousel__figure portfolio-carousel__figure--tablet"
      >
        <img
          alt=""
          className="portfolio-carousel__device-frame"
          src="frame-tablet.svg"
        />
        <div
          className="portfolio-carousel__device-img portfolio-carousel__device-img--tablet"
        >
          <img
            alt=""
            src="http://ossify.com/spectacularism/zigzagged?a=communalization&b=sharky#nephrogastric"
          />
        </div>
      </figure>
    </div>
    <div
      className="portfolio-carousel__slide portfolio-carousel__slide--desktop"
    >
      <figure
        className="portfolio-carousel__figure portfolio-carousel__figure--desktop"
      >
        <img
          alt=""
          className="portfolio-carousel__device-frame"
          src="frame-laptop.svg"
        />
        <div
          className="portfolio-carousel__device-img portfolio-carousel__device-img--desktop"
        >
          <img
            alt=""
            src="http://yellowy.com/azocochineal/sourjack?a=overcomplete&b=phlegmless#tempi"
          />
        </div>
      </figure>
    </div>
  </Carousel>
</div>
`;
